mods/persistence/modules/modular_computers/file_system/programs/trade_beacon_program.dm

```diff
@@ -0,0 +1,560 @@
+				var/list/accesses = computer.get_access(user)
+				if(net && net.banking_mainframe)
+					if(!net.banking_mainframe.has_access(accesses))
+						data["network_account_error"] = "No access."
+					else
+						var/datum/money_account/parent/network/bank_account = net.banking_mainframe.get_parent_account()
+						if(!bank_account)
+							data["network_account_error"] = "No Network Account"
+						else
+							data["network_account"] = bank_account.account_name
+							data["network_account_balance"] = bank_account.money
+				else
+					data["network_account_error"] = "No Network Bank."
+
+
+	else
+		data["beacon_name"] = "*DISCONNECTED*"
+
+
+	ui = SSnano.try_update_ui(user, src, ui_key, ui, data, force_open)
+	if (!ui)
+		ui = new(user, src, ui_key, "trade_management.tmpl", name, 600, 750, state = state)
+		ui.auto_update_layout = 1
+		ui.set_initial_data(data)
+		ui.set_auto_update(1)
+		ui.open()
+
+/datum/computer_file/program/trade_management/Topic(href, href_list, state)
+	. = ..()
+	if(.)
+		return
+
+	if(href_list["select_beacon"])
+		var/obj/hold = computer.get_physical_host()
+		var/list/possible_trade_beacons = hold.get_adjacent_trade_beacons()
+		possible_trade_beacons += "*DISCONNECT*"
+		var/beacon = input(usr, "Select a nearby trade beacon.", "Pick a beacon") as null|anything in possible_trade_beacons
+		if(!CanInteract(usr, global.default_topic_state))
+			return TOPIC_REFRESH
+		if(beacon == "*DISCONNECT*")
+			selected_beacon = null
+		else if(beacon)
+			selected_beacon = beacon
+
+		return TOPIC_REFRESH
+	if(href_list["import"])
+		import_menu = 1
+		export_menu = 0
+		log_menu = 0
+		selected_import = null
+		selected_export = null
+		return TOPIC_REFRESH
+	if(href_list["export"])
+		import_menu = 0
+		export_menu = 1
+		log_menu = 0
+		selected_import = null
+		selected_export = null
+		return TOPIC_REFRESH
+	if(href_list["log"])
+		import_menu = 0
+		export_menu = 0
+		log_menu = 1
+		selected_import = null
+		selected_export = null
+		return TOPIC_REFRESH
+	if(href_list["select_import"])
+		import_menu = 0
+		export_menu = 0
+		log_menu = 0
+		selected_import = locate(href_list["select_import"])
+		selected_export = null
+		return TOPIC_REFRESH
+	if(href_list["select_export"])
+		import_menu = 0
+		export_menu = 0
+		log_menu = 0
+		selected_import = null
+		selected_export = locate(href_list["select_export"])
+		return TOPIC_REFRESH
+	if(href_list["select_personal_account"])
+		payment_type = 1
+		return TOPIC_REFRESH
+	if(href_list["select_charge_stick"])
+		payment_type = 2
+		return TOPIC_REFRESH
+	if(href_list["select_network_account"])
+		payment_type = 3
+		return TOPIC_REFRESH
+	if(href_list["select_telepad"])
+		detect_telepads()
+		var/obj/T = locate(href_list["select_telepad"])
+		if(T && (T in detected_telepads))
+			selected_telepad = T
+		return TOPIC_REFRESH
+	if(href_list["refresh_telepad"])
+		detect_telepads()
+		return TOPIC_REFRESH
+	if(href_list["buy_import"])
```

Comment:
You really want to move the code under "buy_import" and "buy_export" into their own procs, because all those indents get particularly hard to read on this particular case 0_o

Reply:
fixed